rest/route/aws.go

```diff
@@ -0,0 +1,99 @@
+package route
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	sns "github.com/robbiet480/go.sns"
+)
+
+const (
+	messageTypeSubscriptionConfirmation = "SubscriptionConfirmation"
+	messageTypeNotification             = "Notification"
+	messageTypeUnsubscribeConfirmation  = "UnsubscribeConfirmation"
+)
+
+type awsSns struct {
+	sc          data.Connector
+	messageType string
+	payload     sns.Payload
+}
+
+func makeAwsSnsRoute(sc data.Connector) gimlet.RouteHandler {
+	return &awsSns{
+		sc: sc,
+	}
+}
+
+func (aws *awsSns) Factory() gimlet.RouteHandler {
+	return &awsSns{
+		sc: aws.sc,
+	}
+}
+
+func (aws *awsSns) Parse(ctx context.Context, r *http.Request) error {
+	aws.messageType = r.Header.Get("x-amz-sns-message-type")
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return errors.Wrap(err, "problem reading body")
+	}
+	if err = json.Unmarshal(body, &aws.payload); err != nil {
+		return errors.Wrap(err, "problem unmarshalling payload")
+	}
+
+	if err = aws.payload.VerifyPayload(); err != nil {
+		msg := "AWS SNS message failed validation"
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": msg,
+			"payload": aws.payload,
+		}))
+		return errors.Wrap(err, msg)
+	}
+
+	return nil
+}
+
+func (aws *awsSns) Run(ctx context.Context) gimlet.Responder {
+	switch aws.messageType {
+	case messageTypeSubscriptionConfirmation:
```

Comment:
Your PR description was really helpful! It might be good to translate some of that into comments for future engineers, so they don't have to look up the PR to understand what's going on here. (For example, explaining that Subscription/Unsubscription is a rare action that we will handle manually and will be logged  to splunk given the logging level)

Reply:
done